<launch>
  <!--############### Node Configurations ##################-->
  <arg name="output" default="screen"/>
  <arg name="fps"    default="5"/>
  <arg name="shutter_ms"    default="2.0"/>

  <!-- Camera Configurations -->
  <arg name="cam0" default="17041930"/>
  <arg name="cam1" default="17041931"/>
  <arg name="cam2" default="17041932"/>
  <arg name="cam3" default="17041925"/>

  <arg name="duration" default="30000" />
  <arg name="num_cameras" default="4" />

  <!-- Broadcast camera image for specified amount of time. -->
  <node pkg="poll_cameras" type="poll_cameras_node" name="poll_cameras" output="$(arg output)">
    <param name="cam0/identifier" type="string" value="$(arg cam0)"/>
    <param name="cam1/identifier" type="string" value="$(arg cam1)"/>
    <param name="cam2/identifier" type="string" value="$(arg cam2)"/>
    <param name="cam3/identifier" type="string" value="$(arg cam3)"/>

    <param name="cam0/camera_name" type="string" value="pg_$(arg cam0)"/>
    <param name="cam1/camera_name" type="string" value="pg_$(arg cam1)"/>
    <param name="cam2/camera_name" type="string" value="pg_$(arg cam2)"/>
    <param name="cam3/camera_name" type="string" value="pg_$(arg cam3)"/>

    <param name="cam0/calib_url" type="string" value="file://${ROS_HOME}/camera_info/pg_$(arg cam0).yaml"/>
    <param name="cam1/calib_url" type="string" value="file://${ROS_HOME}/camera_info/pg_$(arg cam1).yaml"/>
    <param name="cam2/calib_url" type="string" value="file://${ROS_HOME}/camera_info/pg_$(arg cam2).yaml"/>
    <param name="cam3/calib_url" type="string" value="file://${ROS_HOME}/camera_info/pg_$(arg cam3).yaml"/>

    <param name="rec_length" type="double" value="$(arg duration)"/>
    <param name="num_cameras" type="int" value="$(arg num_cameras)"/>
    <param name="fps" type="double" value="$(arg fps)"/>
    <!-- format7_mode
	 0 = full resolution,
	 1 = 2x2 bayer binning (1 x fps, half resolution, but 4 x brightness),
	 2 = 2x2 decimation (2 x fps, half resolution, 1 x brightness)
    -->
    <param name="format7_mode" type="int" value="1"/>
    <param name="width"  type="int" value="1024"/>
    <param name="height" type="int" value="1024"/>
    <param name="auto_white_balance" type="bool" value="false"/>
    <param name="white_balance" type="bool" value="true"/>
    <param name="wb_red" type="int" value="652"/>
    <param name="wb_blue" type="int" value="589"/>
    
    <param name="shutter_ms" type="double" value="$(arg shutter_ms)"/>
    <param name="gain_db" type="double" value="0"/>
  </node>
</launch>
